uktrade/data-hub-frontend
# Push referral info to Google Tag Manager when sending a referral

## Description of change

This PR pushes information about a referral to the Google Tag Manager 'dataLayer', to be used in Google Analytics, on submit of the send referral form. This will allow more in-depth analysis of the teams who are referring companies to each other and the content of referrals. 

A new saga has been created in order to do this, as the `Task` saga is designed for asynchronous calls and returns a promise which is not necessary for pushing to the data layer. As there was previously only one saga passed to `sagaMiddleware`, a `rootSaga` has been created which combines multiple sagas and then is passed to `sagaMiddleware`.

The component `TagManagerDataLayer` is designed to take components which trigger a push to the data layer as children, and passes its children a function which fires the appropriate action. This approach was chosen rather than having each component dispatch the `TAG_MANAGER_DATA_LAYER__PUSH` action itself, as it removes the need for these components to connect to the redux store unnecessarily. 

Only the team name of the sending and receiving advisers is pushed to Tag Manager to avoid any personal identifiable information being sent. In addition to the subject of the referral. 

## Test instructions

If you have access to Tag Manager:
- Go into preview mode
- Go to a company
- Fill out the send referral form
- Cmd or ctrl click on 'Send referral'
- Look in the 'Data Layer' tab of the Tag Manager preview window
- Check that the correct data is under the 'send_referral' event

## Screenshots

No visual changes

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
